ext/nnstreamer/tensor_filter/tensor_filter_llamacpp.cc

```diff
@@ -214,26 +243,61 @@ TensorFilterLlamaCpp::invoke (const GstTensorMemory *input, GstTensorMemory *out
 }
 
 /**
- * @brief Invoke llamacpp using input tensors
+ * @brief Create output tensor
+ *
+ * The memory allocated for the output tensor data using strndup
+ * must be freed by the caller of this function to avoid memory leaks.
+ */
+GstTensorMemory *
+TensorFilterLlamaCpp::createOutputTensor (
+    const char *buf, int n, GstTensorFilterProperties *prop)
+{
+  if (buf == nullptr || n <= 0 || prop == nullptr) {
+    g_critical ("Invalid arguments passed to createOutputTensor");
+    return nullptr;
+  }
+
+  GstTensorMemory *output = g_new0 (GstTensorMemory, 1);
+  if (!output) {
+    g_critical ("createOutputTensor: Allocation failed");
+    return nullptr;
+  }
+
+  std::string s (buf, n);
```

Comment:
is it necessary?
```
output[0].data = g_strndup (buf, n);
output[0].size = n + 1;
```

Reply:
Yes.
std::string s(buf, n); // Use first n bytes from buf (may not include NULL)
s.length() == n;       // String length excludes NULL
strndup(s.c_str(), n); // Duplicate n bytes, auto appends NULL
output[0].size = n;    // Stores string length, NULL excluded